pkg/virt-controller/watch/vm_test.go

```diff
@@ -5316,6 +5379,301 @@ var _ = Describe("VirtualMachine", func() {
+				k8sClient.Fake.PrependReactor("create", "controllerrevisions", func(action testing.Action) (handled bool, obj runtime.Object, err error) {
+					created, ok := action.(testing.CreateAction)
+					Expect(ok).To(BeTrue())
+
+					createObj, ok := created.GetObject().(*appsv1.ControllerRevision)
+					Expect(ok).To(BeTrue())
+
+					crListLock.Lock()
+					defer crListLock.Unlock()
+					crList.Items = append(crList.Items, *createObj)
+					crSource.Add(createObj)
+
+					return true, created.GetObject(), nil
+				})
+			}
+
+			crFor := func(uid string) string {
+				crListLock.Lock()
+				defer crListLock.Unlock()
+				for _, cr := range crList.Items {
+					if strings.Contains(cr.Name, uid) {
+						return cr.Name
+					}
+				}
+
+				return ""
+			}
+
+			BeforeEach(func() {
+				k8sClient.Fake.ClearActions()
+				crList = appsv1.ControllerRevisionList{}
+				vm, vmi = DefaultVirtualMachine(true)
+				vm.ObjectMeta.UID = types.UID(uuid.NewString())
+				vmi.ObjectMeta.UID = vm.ObjectMeta.UID
+				vm.Generation = 1
+				vm.Spec.Template.Spec.Domain.CPU = &v1.CPU{
+					Cores: 2,
+				}
+				guest := resource.MustParse("128Mi")
+				vm.Spec.Template.Spec.Domain.Memory = &v1.Memory{
+					Guest: &guest,
+				}
+				kv = &v1.KubeVirt{
+					Spec: v1.KubeVirtSpec{
+						Configuration: v1.KubeVirtConfiguration{
+							LiveUpdateConfiguration: &virtv1.LiveUpdateConfiguration{},
+							VMRolloutStrategy: &v1.VMRolloutStrategy{
+								LiveUpdate: &virtv1.RolloutStrategyLiveUpdate{},
+							},
+							DeveloperConfiguration: &v1.DeveloperConfiguration{
+								FeatureGates: []string{virtconfig.VMLiveUpdateFeaturesGate},
+							},
+						},
+					},
+				}
+				restartRequired = false
+			})
+
+			AfterEach(func() {
+				k8sClient.Fake.ClearActions()
+			})
+
+			It("should appear when changing a non-live-updatable field", func() {
+				testutils.UpdateFakeKubeVirtClusterConfig(kvInformer, kv)
+
+				By("Creating a VM with hostname a")
+				vm.Spec.Template.Spec.Hostname = "a"
+				addVirtualMachine(vm)
+
+				By("Executing the controller expecting a VMI to get created and no RestartRequired condition")
+				vmi = controller.setupVMIFromVM(vm)
+				expectVMICreation()
+				expectVMStatusUpdate()
+				expectControllerRevisionList()
+				expectControllerRevisionCreation()
+				controller.Execute()
+				Expect(cache.WaitForCacheSync(stop, crInformer.HasSynced)).To(BeTrue())
+				Consistently(restartRequired, 1*time.Second).Should(BeFalse(), "restart required")
+				markAsReady(vmi)
+				vmiFeeder.Add(vmi)
+
+				By("Bumping the VM sockets above the cluster maximum")
+				vm.Spec.Template.Spec.Hostname = "b"
+				vm.Generation = 2
+				modifyVirtualMachine(vm)
+
+				By("Executing the controller again expecting the RestartRequired condition to appear")
+				expectControllerRevisionDelete()
+				expectVMUpdate()
+				controller.Execute()
+				Expect(cache.WaitForCacheSync(stop, crInformer.HasSynced)).To(BeTrue())
```

Comment:
Can be dropped

Reply:
I think this one makes sense, it is just improved by the commit that simplifies sync.